src/thread.rs

```diff
@@ -509,6 +519,21 @@ impl<'mir, 'tcx: 'mir> ThreadManager<'mir, 'tcx> {
             throw_machine_stop!(TerminationInfo::Deadlock);
         }
     }
+
+    /// Store the panic payload when beginning unwinding.
+    fn set_panic_payload(&mut self, payload: Scalar<Tag>) {
+        let thread = self.active_thread_mut();
+        assert!(
+            thread.panic_payload.is_none(),
+            "the panic runtime should avoid double-panics"
+        );
+        thread.panic_payload = Some(payload);
+    }
```

Comment:
This is moving panic logic into `thread.rs`. That file already contains too much logic IMO, I'd rather avoid this.

Is it possible to structure this differently, make the `panic_payload` field `pub(crate)` as before, and have `panic.rs` do the "get active thread, then work on its panic payload" logic?